torch/csrc/utils/pybind.h

```diff
@@ -1,33 +1,44 @@
 #pragma once
 
 #include <Python.h>
+
 #include <ATen/ATen.h>
 #include <pybind11/pybind11.h>
 #include <pybind11/stl.h>
 
 #include "torch/csrc/DynamicTypes.h"
 #include "torch/csrc/autograd/python_variable.h"
 
+#include <stdexcept>
+
 namespace py = pybind11;
 
 namespace pybind11 { namespace detail {
 
-// handle Tensor <-> at::Tensor conversions
-// Python Variables are unpacked into Tensors
-template <> struct type_caster<at::Tensor> {
-public:
+// torch.autograd.Variable <-> at::Tensor conversions (without unwrapping)
+template <>
+struct type_caster<at::Tensor> {
+ public:
   PYBIND11_TYPE_CASTER(at::Tensor, _("at::Tensor"));
 
   bool load(handle src, bool) {
     PyObject* obj = src.ptr();
     if (THPVariable_Check(obj)) {
-      value = ((THPVariable*)obj)->cdata.data();
+      value = reinterpret_cast<THPVariable*>(obj)->cdata;
```

Comment:
This needs a very careful review, because it's used in a few places in C++ that this diff hasn't touched.

Reply:
Note: I put a `throw runtime_error()` here to catch all visits of this code and fixed the places that reached it (while running all tests)